### PR TITLE
feat(rule): get collection values from RULES_METADATA event [CARROT-1615]

### PR DESCRIPTION
**Pull request by AMarcosCastelo**
### Summary

This PR refactors the NFT metadata selection functionality by consolidating the metadata event value handling into a single, more robust function. The changes improve type safety and test coverage while simplifying the codebase.

### Details

Key changes:
1. Consolidated metadata handling:
   - Removed `getImageFromMetadata` function and merged its functionality into `getRulesMetadataEventValues`
   - Added proper type assertions for collection name and description 
   - Improved error handling for missing or invalid metadata values

2. Test improvements:
   - Updated tests to use proper stubbing patterns with `stubDocument` and `stubDocumentEventWithMetadataAttributes`
   - Added test cases for collection name/description validation
   - Added comprehensive error case coverage
   - Fixed type safety issues in test implementation

3. Code organization:
   - Maintained consistent use of `DocumentEventAttributeName` constants
   - Improved type safety with proper assertions
   - Better error handling for missing required attributes

The changes make the code more maintainable and robust while ensuring proper validation of NFT metadata values.

### Related links

https://app.clickup.com/t/86a5rmahh

---

- [x] **I, the PR author, declare that this PR works as expected and does not break any service. I also declare that I gave my best to apply all of the best practices and that I'm leaving the code better than I found it.**

## Summary by CodeRabbit

- **New Features**
	- Enhanced NFT metadata handling with the addition of `collectionName` and `nftDescription` attributes.
	- Updated functions to support the new metadata attributes, improving the flexibility of NFT representations.

- **Bug Fixes**
	- Improved error handling in metadata extraction processes to ensure robustness against invalid inputs.

- **Tests**
	- Comprehensive updates to the test suite to validate the new metadata attributes and their handling in various scenarios.
	- Enhanced test cases for accurate validation of NFT metadata extraction with new attributes.

### CODE REVIEW

**Review by GLGuilherme — DISMISSED** [review]

File: apps/methodologies/bold/rule-processors/credit/nft-metadata-selection/src/lib/nft-metadata-selection.helpers.ts [hunk 1]
```diff
@@ -254,6 +275,8 @@ export const mapNftMetadataDto = (
   });
 
 export const mapNftMetadata = ({
+  collectionDescription,
+  collectionName,
   creditDocumentId,
```
[the remark this answers is not in the document]
Comment (AMarcosCastelo) [reply apps/methodologies/bold/rule-processors/credit/nft-metadata-selection/src/lib/nft-metadata-selection.helpers.ts]: [feature/handle-collection-data 8741c96] refactor: change COLLECTION_DESCRIPTION to NFT_DESCRIPTION

File: apps/methodologies/bold/rule-processors/credit/nft-metadata-selection/src/lib/nft-metadata-selection.helpers.ts [hunk 2]
```diff
@@ -43,22 +47,39 @@
...
+    rulesMetadataEvent,
+    COLLECTION_DESCRIPTION,
+  );
 
-  return is<Uri>(uri) ? uri : undefined;
+  return {
+    collectionDescription: assert<NonEmptyString>(collectionDescription),
+    collectionName: assert<NonEmptyString>(collectionName),
+    image: is<Uri>(uri) ? uri : undefined,
+  };
```
[the remark this answers is not in the document]
Comment (AMarcosCastelo) [reply apps/methodologies/bold/rule-processors/credit/nft-metadata-selection/src/lib/nft-metadata-selection.helpers.ts]: [feature/handle-collection-data 7799229] refactor: handle errors in getRulesMetadataEventValues helper

File: apps/methodologies/bold/rule-processors/credit/nft-metadata-selection/src/lib/nft-metadata-selection.processor.e2e.spec.ts [hunk 3]
```diff
@@ -129,6 +131,14 @@ describe('NftMetadataSelection E2E', () => {
...
               value: image,
             }),
+            stubDocumentEventAttribute({
+              name: COLLECTION_NAME,
+              value: faker.lorem.word(),
+            }),
+            stubDocumentEventAttribute({
+              name: COLLECTION_DESCRIPTION,
+              value: faker.lorem.sentence(),
+            }),
```
[the remark this answers is not in the document]
Comment (AMarcosCastelo) [reply apps/methodologies/bold/rule-processors/credit/nft-metadata-selection/src/lib/nft-metadata-selection.processor.e2e.spec.ts]: [feature/handle-collection-data 941cd41] test: add an e2e assertion to verify the collectionName and nftDescription

File: apps/methodologies/bold/rule-processors/credit/nft-metadata-selection/src/lib/nft-metadata-selection.helpers.ts [hunk 4]
```diff
@@ -43,22 +47,51 @@ import type {
...
+    COLLECTION_NAME,
+  );
+  const nftDescription = getEventAttributeValue(
+    rulesMetadataEvent,
+    NFT_DESCRIPTION,
+  );
 
-  return is<Uri>(uri) ? uri : undefined;
+  if (!is<NonEmptyString>(collectionName)) {
+    throw new Error('Required metadata collectionName attribute is missing');
```
Comment (cris-santos) [inline apps/methodologies/bold/rule-processors/credit/nft-metadata-selection/src/lib/nft-metadata-selection.helpers.ts]:
```suggestion
    throw new Error(`Required metadata ${COLLECTION_NAME} attribute is missing`);
```

File: apps/methodologies/bold/rule-processors/credit/nft-metadata-selection/src/lib/nft-metadata-selection.helpers.ts [hunk 5]
```diff
@@ -43,22 +47,51 @@ import type {
...
+  }
+
+  if (!is<NonEmptyString>(nftDescription)) {
+    throw new Error('Required metadata nftDescription attribute is missing');
```
Comment (cris-santos) [inline apps/methodologies/bold/rule-processors/credit/nft-metadata-selection/src/lib/nft-metadata-selection.helpers.ts]:
```suggestion
    throw new Error(`Required metadata ${NFT_DESCRIPTION} attribute is missing`);
```

**Review by cris-santos — DISMISSED** [review]

**Review by GLGuilherme — DISMISSED** [review]

File: apps/methodologies/bold/rule-processors/credit/nft-metadata-selection/src/lib/nft-metadata-selection.helpers.ts (hunk 4 above)
Comment (AMarcosCastelo) [reply apps/methodologies/bold/rule-processors/credit/nft-metadata-selection/src/lib/nft-metadata-selection.helpers.ts]: [feature/handle-collection-data 3b0c559] refactor: use enum values to create error messages

File: apps/methodologies/bold/rule-processors/credit/nft-metadata-selection/src/lib/nft-metadata-selection.helpers.ts (hunk 5 above)
Comment (AMarcosCastelo) [reply apps/methodologies/bold/rule-processors/credit/nft-metadata-selection/src/lib/nft-metadata-selection.helpers.ts]: [feature/handle-collection-data 3b0c559] refactor: use enum values to create error messages

**Review by cris-santos — APPROVED** [review]

**Review by GLGuilherme — APPROVED** [review]

**Pull request merged**